services/121-service/src/events/events.service.ts

```diff
@@ -259,19 +269,27 @@ export class EventsService {
 
     const events: EventEntity[] = [];
     for (const fieldName of fieldNames) {
+      const key = fieldName as keyof LogEntity;
+
       if (
-        oldEntity[fieldName] === newEntity[fieldName] ||
-        (isObject(oldEntity[fieldName]) &&
-          isObject(newEntity[fieldName]) &&
-          isMatch(oldEntity[fieldName], newEntity[fieldName]))
+        oldEntity[key] === newEntity[key] ||
+        (isObject(oldEntity[key]) &&
+          isObject(newEntity[key]) &&
+          isMatch(oldEntity[key], newEntity[key]))
       ) {
         continue;
       }
 
+      // Ensure only string values are passed to the event creation
+      const oldValue =
+        typeof oldEntity[key] === 'string' ? oldEntity[key] : null;
+      const newValue =
+        typeof newEntity[key] === 'string' ? newEntity[key] : null;
```

Comment:
Maybe we should throw an error instead, because otherwise we are silently "failing"

```suggestion
      if (typeof oldEntity[key] !== 'string') { throw Error('something something'); }
      if (typeof newEntity[key] !== 'string') { throw Error('something something'); }
```

Or could the `createEventForChange` method also support different types? Maybe also numbers?

Reply:
@aberonni  What do you think about this: 
const oldValue =
        typeof oldEntity[key] === 'string'
          ? oldEntity[key]
          : (() => {
              throw Error('something something');
            })();
      const newValue =
        typeof newEntity[key] === 'string'
          ? newEntity[key]
          : (() => {
              throw Error('something something');
            })();

Reply:
I think maybe this is cleaner:


```suggestion
      if (typeof oldEntity[key] !== 'string') { throw Error('something something'); }
      if (typeof newEntity[key] !== 'string') { throw Error('something something'); }
      
      const oldValue = oldEntity[key] as string;
      const newValue = newEntity[key] as string;
```

PS if you wrap your text with ``` it makes it a lot easier to read, something like:

<img width="418" alt="image" src="https://github.com/user-attachments/assets/9211756b-1937-4a3b-98f2-dddd6e1a51c1">

Will render like this:

```ts
const oldValue =
        typeof oldEntity[key] === 'string'
          ? oldEntity[key]
          : (() => {
              throw Error('something something');
            })();
      const newValue =
        typeof newEntity[key] === 'string'
          ? newEntity[key]
          : (() => {
              throw Error('something something');
            })();
```

Reply:
okay, I will change it.